OnBoardingCalculator/app/src/main/java/com/globant/onboardingcalculator/mvp/presenter/CalculatorPresenter.java

```diff
@@ -20,18 +33,74 @@ public void onClearPressed() {
             view.refreshVisor(model.getSecondOperand());
         }
     }
-}
\ No newline at end of file
+
+    public void onEqualPressed() {
+        if (!model.emptyOperation())
+            switch (model.getOperator()) {
+                case OPERATOR_PLUS:
+                    model.operate(String.valueOf(decimalFormat.format(parseDouble(model.getFirstOperand()) + parseDouble(model.getSecondOperand()))));
+                    break;
+                case OPERATOR_SUBSTRACTION:
+                    String.valueOf(decimalFormat.format(parseDouble(model.getFirstOperand()) - parseDouble(model.getSecondOperand())));
+                    model.operate(String.valueOf(decimalFormat.format(parseDouble(model.getFirstOperand()) - parseDouble(model.getSecondOperand()))));
+                    break;
+                case OPERATOR_MULTIPLY:
+                    model.operate(String.valueOf(decimalFormat.format(parseDouble(model.getFirstOperand()) * parseDouble(model.getSecondOperand()))));
+                    break;
+                case OPERATOR_DIVIDE:
+                    if (parseDouble(model.getSecondOperand()) != parseDouble(NUMBER_ZERO)) {
+                        model.operate(String.valueOf(decimalFormat.format(parseDouble(model.getFirstOperand()) / parseDouble(model.getSecondOperand()))));
+                    } else {
+                        model.clearOperation();
+                        view.showMathError();
+                        break;
+                    }
+            }
+        view.refreshVisor(model.getResult());
+    }
+
+    public void onPointPressed() {
+        if (!model.getResult().equals(EMPTY_STRING) && (model.getOperator() == EMPTY_CHAR))
+            view.showOperatorErrorAfterEqualPressed();
+        else if (model.getOperator() == EMPTY_CHAR) {
+                if (model.getFirstOperand().isEmpty()) {
+                    model.setFirstOperand(NUMBER_ZERO);
+                    model.setFirstOperand(DECIMAL_POINT);
+                    view.refreshVisor(model.getFirstOperand());
+                } else if (model.getFirstOperand().contains(DECIMAL_POINT))
+                            view.showDecimalError();
+                        else {
+                            model.setFirstOperand(DECIMAL_POINT);
+                            view.refreshVisor(model.getFirstOperand());
+                }
+            } else if (model.getSecondOperand().isEmpty()) {
+                        model.setSecondOperand(NUMBER_ZERO);
+                        model.setSecondOperand(DECIMAL_POINT);
+                        view.refreshVisor(model.getSecondOperand());
+                    } else if (model.getSecondOperand().contains(DECIMAL_POINT))
+                                view.showDecimalError();
+                            else {
+                                model.setSecondOperand(DECIMAL_POINT);
+                                view.refreshVisor(model.getSecondOperand());
+                            }
```

Comment:
this code is repeated at line 89, you can create a function onOperandContainsDecimalPoint(model.operand)

Reply:
Nico, I talked with Mau about that and I made a little refactor of this method.